compare.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"golang.org/x/perf/benchstat"
+)
+
+const BenchstatPrefix = "Benchmark"
+
+// Compare compares runs from the same platform
+func Compare(s []string) {
+	builders := make(map[string]*strings.Builder)
+	for _, path := range s {
+		info, err := os.Stat(path)
+		if err != nil {
+			panic(err)
+		}
+
+		if !info.Mode().IsDir() {
+			panic(fmt.Errorf("[compare] %s is not a valid path", path))
+		}
+
+		files, err := ioutil.ReadDir(path)
+		if err != nil {
+			panic(err)
+		}
+
+		for _, f := range files {
+			if f.IsDir() {
+				name := f.Name()
+				p := filepath.Join(path, name)
+				if b, ok := builders[name]; ok {
+					addResult(b, p)
+				} else {
+					var sb strings.Builder
+					builders[name] = &sb
+					addResult(&sb, p)
+				}
+			}
+		}
+	}
+
+	c := &benchstat.Collection{
+		Alpha:      0.05,
+		AddGeoMean: false,
+		DeltaTest:  benchstat.UTest,
+	}
+
+	for key, b := range builders {
+		if err := c.AddFile(key, strings.NewReader(b.String())); err != nil {
```

Comment:
benchstat can take an arbitrary amount of comparisons, hence the approach to use a map of builders (in memory strings that contain values from `instrumented`, `baseline`, etc.)

Should we write these files somewhere?